modules/brainbrowser/js/brainbrowser.loris.js

```diff
@@ -740,71 +740,79 @@ $(function() {
-              }
-            });
-          } else if (type === "application/x-nii") {
-            minc_volumes.push({
-              type: 'nifti1',
-              nii_url: loris.BaseURL + "/brainbrowser/ajax/image.php?file_id=" + fileid,
-              template: {
-                element_id: "volume-ui-template4d",
-                viewer_insert_class: "volume-viewer-display"
-              }
-            });
-          } else {
-            let msg = "WARNING: The volume viewer only supports MINC and" +
-                      " NIfTI file types.";
-            $("#loading").html(msg);
-            console.group('warning message');
+    function requestMINC(minc_id) {
+      return new Promise((resolve) => {
+        $.ajax({
+          url: loris.BaseURL + "/brainbrowser/ajax/image.php",
+          data: 'file_id=' + minc_id,
+          method: 'GET',
+          success: function (response, status, jqXHR) {
+            let type = jqXHR.getResponseHeader('Content-Type');
+            let fileid = jqXHR.getResponseHeader('X-FileID');
+            if (type === "application/x-mnc") {
+              minc_volumes.push({
+                type: 'minc',
+                raw_data_url: loris.BaseURL + "/brainbrowser/ajax/image.php?file_id=" + fileid,
+                template: {
+                  element_id: "volume-ui-template4d",
+                  viewer_insert_class: "volume-viewer-display"
+                }
+              });
+            } else if (type === "application/x-nii") {
+              minc_volumes.push({
+                type: 'nifti1',
+                nii_url: loris.BaseURL + "/brainbrowser/ajax/image.php?file_id=" + fileid,
+                template: {
+                  element_id: "volume-ui-template4d",
+                  viewer_insert_class: "volume-viewer-display"
+                }
+              });
+            } else {
+              let msg = "WARNING: The volume viewer only supports MINC and" +
+                " NIfTI file types.";
+              $("#loading").html(msg);
+              console.group('warning message');
               console.warn(msg);
               console.warn("\nNot supported file was " + filename);
-            console.groupEnd();
-            console.log($("#loading").text());
-          }
-        },
-        error: function(jqXHR, status, errorThrown) {
-          if (errorThrown === "Not Found") {
-            let msg = "ERROR: file not found.";
-            $("#loading").html(msg);
-            console.error(msg);
+              console.groupEnd();
+              console.log($("#loading").text());
+            }
+            resolve();
+          },
+          error: function (jqXHR, status, errorThrown) {
+            console.log('error');
+            if (errorThrown === "Not Found") {
+              let msg = "ERROR: file not found.";
+              $("#loading").html(msg);
+              console.error(msg);
+            }
+            resolve();
```

Comment:
shouldn't this reject the promise if there's an error?

Reply:
@driusan I asked @HenriRabalais and we thought it would be best to not do that because it would result in the other brains not loading. We tjpigjt it would be best to work on the error handling in another PR when the jquery ajax is removed to be fetch.

Reply:
@cmadjar thoughts on this from a user perspective? If there's an error with one should the others still show?

Reply:
I don't really have a strong opinion on this. I see pros and cons on both sides...

Reply:
Okay, then once the whitespace is fixed and it's passing Travis I'll merge it as-is and it can be revisited later.